Issuer/Issuer.cs

```diff
@@ -45,10 +43,51 @@ public static string Name()
             return ISSUER_NAME;
         }
 
-        public static void _deploy(object data, bool update)
+        [DisplayName("_deploy")]
+        public static void OnDeploy(object data, bool update)
         {
             if (update) return;
-            AdminList.Put(ISSUER_DEFAULT_PUBLIC_KEY, 0);
+
+            var tx = (Transaction)Runtime.ScriptContainer;
+            var key = new byte[] { Prefix_Owner };
+            Storage.Put(Storage.CurrentContext, key, tx.Sender);
+
+            // For now, I'm leaving the AdminList empty on deploy. 
+            // Owner can always call AddAdmin after deployment.
+            
+            // Unsure if the hard-coded default public key added in the source contract
```

Comment:
@bettybao1209 Can you tell me about the hard coded owner & public key from the original contract? Were they from the same account? Is it important for the contract logic that the default AdminList contain the public key of the contract owner?

Reply:
the owner is also the admin, who can remove or add admin.

Reply:
So would you say that deploying the Issuer contract should *require* the owner's public key to be provided via the `data` parameter of the `Deploy` operation - which is in turn passed as the `data` parameter of the `_deploy` operation - and that the Issuer contract should explicitly verify that public key matches the contract owner (aka `tx.Sender`) and fail the deployment if they don't match?

Also, does that mean that the owner should NOT be able to remove their own public key from the AdminList via `RemoveAdmin`? I'm thinking we could track the owner public key separate from the AdminList, but check it explicitly in `IsCalledByAdmin` and explicitly add it to the list in `GetAdminList`

Reply:
yes, I think the pubkey of the owner should be passed in by data rather than the scripthash.
Also the owner should not be removed by invoking RemoveAdmin, otherwise he cannot perform functions like `RegisterSchema`, `InjectClaim`, etc.